### PR TITLE
RRRR Corp

### PR DESCRIPTION
**Pull request by Kitsunemitsu**
## About The Pull Request
Adds Rcorp map for the event today.

Adds Rcorp CM and all the roles related.
Cleans up maptype code.
Adds a new gamemode that automatically breaches all abnormalities
If the map requires all abnos to breach it will automatically change the gamemode and breach them.
Adds Auto-possession to game modes.

### CODE REVIEW

File: code/game/objects/items/ego_weapons/non_abnormality/rcorp.dm [hunk 1]
```diff
@@ -34,3 +34,223 @@
...
+			damtype = WHITE_DAMAGE
+		if(WHITE_DAMAGE)
+			damtype = BLACK_DAMAGE
+		if(BLACK_DAMAGE)
+			damtype = PALE_DAMAGE
+		if(PALE_DAMAGE)
+			damtype = RED_DAMAGE
+	armortype = damtype // TODO: In future, armortype should be gone entirely
+	to_chat(user, "<span class='notice'>\The [src] will now deal [damtype] damage.</span>")
+	playsound(src, 'sound/items/screwdriver2.ogg', 50, TRUE)
```
Comment (dsmith328) [inline code/game/objects/items/ego_weapons/non_abnormality/rcorp.dm:65]: This should be inherited entirely from it's parent type (rabbit_blade) and be unnecessary

File: code/game/objects/items/ego_weapons/non_abnormality/rcorp.dm [hunk 2]
```diff
@@ -34,3 +34,223 @@
...
+	if(!istype(target_turf))
+		return
+	if((get_dist(user, target_turf) < 2) || (get_dist(user, target_turf) > 5))
+		return
+	..()
+	ranged_cooldown = world.time + ranged_cooldown_time
+	playsound(target_turf, 'sound/weapons/pulse.ogg', 50, TRUE)
+	for(var/turf/open/T in range(target_turf, 0))
+		new /obj/effect/temp_visual/smash1(T)
+		for(var/mob/living/L in T.contents)
```
Comment (dsmith328) [inline code/game/objects/items/ego_weapons/non_abnormality/rcorp.dm:102]:
```suggestion
		for(var/mob/living/L in T)
```
I've been told `.contents` is implied and therefore redundant.

File: code/game/objects/items/ego_weapons/non_abnormality/rcorp.dm [hunk 3]
```diff
@@ -34,3 +34,223 @@
...
+	ammo_type = list(
+		/obj/item/ammo_casing/energy/laser/red,
+		)
+	can_charge = FALSE
+	weapon_weight = WEAPON_HEAVY // No dual wielding
+	pin = /obj/item/firing_pin
+
+/obj/item/ego_weapon/city/rabbit_rush
+	name = "rush dagger"
+	desc = "A high-frequency combat blade made for use against abnormalities and other threats in Lobotomy Corporation and the outskirts. This only has one mode"
```
Comment (dsmith328) [inline code/game/objects/items/ego_weapons/non_abnormality/rcorp.dm:187]: This one does not have it's abilities listed anywhere. Perhaps add it in `special` as it inherits from `ego_weapon`?

File: icons/obj/clothing/ego_gear/head.dmi
Comment (dsmith328) [inline icons/obj/clothing/ego_gear/head.dmi:1]: Files in here are direction when they shouldn't be.

**Review by dsmith328 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/ego_weapons/non_abnormality/rcorp.dm (hunk 3 above)
Comment (Kitsunemitsu) [reply code/game/objects/items/ego_weapons/non_abnormality/rcorp.dm:187]: Right, I will add that

**Review by dsmith328 — APPROVED** [review]

**Pull request merged**